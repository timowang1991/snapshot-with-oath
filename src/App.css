.App {
  text-align: center;
  max-width: 1260px;
  margin: 20px auto;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

.app-title {
  font-size: 60px;
  font-family: "Helvetica Neue Light";
}

#app-search-title {
  margin-right: 20px;
}

#oath-title {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  color: #000;
}

#colon-title {
  font-family: sans-serif;
  font-weight: bold;
  color: #006CB7;
}

#search-bar {
  margin-top: 30px;
}

#search-result-list-wrapper {
  list-style-type: none;
  text-align: left;
}

.list-item-wrapper {
  margin-top: 30px;
  margin-right: 30px;
  display: inline-block;
  padding: 10px;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, .2);
  vertical-align: top;
}

.list-item-wrapper::after {
  clear: both;
}

.list-item-image-wrapper {
  float: left;
}

.list-item-image {
  height: 150px;
  width: 150px;
  border-radius: 46%;
}

.person-oath-list-wrapper {
  list-style-type: none;
  float: left;
}

.person-oath-list-item {
  font-size: 17.5px;
  font-family: Georgia,serif;
  color: #3a3a3a;
  font-style: italic;
}

.person-oath-list-item-name {
  margin-top: 10px;
  text-align: center;
  font-size: 16px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
