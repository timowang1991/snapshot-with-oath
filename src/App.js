import React, { Component } from 'react';
import { InputGroup, FormControl, Glyphicon } from 'react-bootstrap';
import './App.css';
import searchOathData from './searchOathData';
import PersonOathListItem from './PersonOathListItem';

class App extends Component {
    constructor() {
        super();
        this.state = {
            data: searchOathData()
        };
    }

    getSearchResults = (e) => {
        this.setState({
            data: searchOathData(e.target.value.toLowerCase())
        });
    }

    render() {
        return (
            <div className="App">
                <div className="app-title">
                    <span id="app-search-title">Search Your</span>
                    <span id="oath-title">Oath</span>
                    <span id="colon-title">:</span>
                </div>
                <InputGroup id="search-bar" bsSize="large">
                    <InputGroup.Addon>
                        <Glyphicon glyph="glyphicon glyphicon-search" />
                    </InputGroup.Addon>
                    <FormControl
                        type="text"
                        placeholder="Search by name..."
                        onChange={this.getSearchResults}
                    />
                </InputGroup>
                <ul id="search-result-list-wrapper">
                {
                    this.state.data.map((item, idx) =>
                        <PersonOathListItem key={`PersonOathListItem-${idx}`} {...item} />
                    )
                }
                </ul>
            </div>
        );
    }
}

export default App;
