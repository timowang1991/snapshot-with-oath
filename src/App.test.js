import React from 'react';
import renderer from 'react-test-renderer';
import { render, shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';
import App from './App';

expect.addSnapshotSerializer(enzymeSerializer);

it('renders app with react-test-renderer', () => {
    const app = renderer.create(
        <App />
    );

    expect(app).toMatchSnapshot();
});

it('renders app with render enzyme', () => {
    const app = render(
        <App />
    );

    expect(app).toMatchSnapshot();
});

it('renders app with shallow enzyme', () => {
    const app = shallow(
        <App />
    );

    expect(app).toMatchSnapshot();
});
