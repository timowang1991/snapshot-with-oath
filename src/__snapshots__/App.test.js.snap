// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app with react-test-renderer 1`] = `
<div
  className="App"
>
  <div
    className="app-title"
  >
    <span
      id="app-search-title"
    >
      Search Your
    </span>
    <span
      id="oath-title"
    >
      Oath
    </span>
    <span
      id="colon-title"
    >
      :
    </span>
  </div>
  <span
    className="input-group input-group-lg"
    id="search-bar"
  >
    <span
      className="input-group-addon"
    >
      <span
        className="glyphicon glyphicon-glyphicon glyphicon-search"
      />
    </span>
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Search by name..."
      type="text"
    />
  </span>
  <ul
    id="search-result-list-wrapper"
  >
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="arvin.png"
        />
        <div
          className="person-oath-list-item-name"
        >
          Arvin
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Workout every night
            <br />
          </blockquote>
        </li>
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            I hate reflection
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="cloe.jpg"
        />
        <div
          className="person-oath-list-item-name"
        >
          Cloe
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Shut Timo up
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="john.png"
        />
        <div
          className="person-oath-list-item-name"
        >
          John
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Why so serious?
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="head-icon-6519.png"
        />
        <div
          className="person-oath-list-item-name"
        >
          Kate
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Work from remote
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="kp.jpg"
        />
        <div
          className="person-oath-list-item-name"
        >
          KP
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Banish all bast**ds
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="sonys.jpg"
        />
        <div
          className="person-oath-list-item-name"
        >
          Sonys
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Youcard is MINE!
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="timo.jpg"
        />
        <div
          className="person-oath-list-item-name"
        >
          Timo
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Listen to TA
            <br />
          </blockquote>
        </li>
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Stop global warming
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="tim-armstrong.jpg"
        />
        <div
          className="person-oath-list-item-name"
        >
          Tim Armstrong
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Give up!
            <br />
          </blockquote>
        </li>
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            Take private jets
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      className="list-item-wrapper"
    >
      <div
        className="list-item-image-wrapper"
      >
        <img
          alt=""
          className="list-item-image"
          src="tsai-ing-wen.jpg"
        />
        <div
          className="person-oath-list-item-name"
        >
          Tsai Ing-wen
        </div>
      </div>
      <ul
        className="person-oath-list-wrapper"
      >
        <li
          className="person-oath-list-item"
        >
          <blockquote>
            No more blackouts
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`renders app with render enzyme 1`] = `
<div
  class="App"
>
  <div
    class="app-title"
  >
    <span
      id="app-search-title"
    >
      Search Your
    </span>
    <span
      id="oath-title"
    >
      Oath
    </span>
    <span
      id="colon-title"
    >
      :
    </span>
  </div>
  <span
    class="input-group input-group-lg"
    id="search-bar"
  >
    <span
      class="input-group-addon"
    >
      <span
        class="glyphicon glyphicon-glyphicon glyphicon-search"
      />
    </span>
    <input
      class="form-control"
      placeholder="Search by name..."
      type="text"
    />
  </span>
  <ul
    id="search-result-list-wrapper"
  >
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="arvin.png"
        />
        <div
          class="person-oath-list-item-name"
        >
          Arvin
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Workout every night
            <br />
          </blockquote>
        </li>
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            I hate reflection
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="cloe.jpg"
        />
        <div
          class="person-oath-list-item-name"
        >
          Cloe
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Shut Timo up
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="john.png"
        />
        <div
          class="person-oath-list-item-name"
        >
          John
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Why so serious?
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="head-icon-6519.png"
        />
        <div
          class="person-oath-list-item-name"
        >
          Kate
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Work from remote
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="kp.jpg"
        />
        <div
          class="person-oath-list-item-name"
        >
          KP
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Banish all bast**ds
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="sonys.jpg"
        />
        <div
          class="person-oath-list-item-name"
        >
          Sonys
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Youcard is MINE!
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="timo.jpg"
        />
        <div
          class="person-oath-list-item-name"
        >
          Timo
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Listen to TA
            <br />
          </blockquote>
        </li>
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Stop global warming
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="tim-armstrong.jpg"
        />
        <div
          class="person-oath-list-item-name"
        >
          Tim Armstrong
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Give up!
            <br />
          </blockquote>
        </li>
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            Take private jets
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
    <li
      class="list-item-wrapper"
    >
      <div
        class="list-item-image-wrapper"
      >
        <img
          alt=""
          class="list-item-image"
          src="tsai-ing-wen.jpg"
        />
        <div
          class="person-oath-list-item-name"
        >
          Tsai Ing-wen
        </div>
      </div>
      <ul
        class="person-oath-list-wrapper"
      >
        <li
          class="person-oath-list-item"
        >
          <blockquote>
            No more blackouts
            <br />
          </blockquote>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`renders app with shallow enzyme 1`] = `
<div
  className="App"
>
  <div
    className="app-title"
  >
    <span
      id="app-search-title"
    >
      Search Your
    </span>
    <span
      id="oath-title"
    >
      Oath
    </span>
    <span
      id="colon-title"
    >
      :
    </span>
  </div>
  <InputGroup
    bsClass="input-group"
    bsSize="large"
    id="search-bar"
  >
    <InputGroupAddon
      bsClass="input-group-addon"
    >
      <Glyphicon
        bsClass="glyphicon"
        glyph="glyphicon glyphicon-search"
      />
    </InputGroupAddon>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      onChange={[Function]}
      placeholder="Search by name..."
      type="text"
    />
  </InputGroup>
  <ul
    id="search-result-list-wrapper"
  >
    <PersonOathListItem
      info={
        Object {
          "image": "arvin.png",
          "oath": Array [
            "Workout every night",
            "I hate reflection",
          ],
        }
      }
      name="Arvin"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "cloe.jpg",
          "oath": Array [
            "Shut Timo up",
          ],
        }
      }
      name="Cloe"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "john.png",
          "oath": Array [
            "Why so serious?",
          ],
        }
      }
      name="John"
    />
    <PersonOathListItem
      info={
        Object {
          "oath": Array [
            "Work from remote",
          ],
        }
      }
      name="Kate"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "kp.jpg",
          "oath": Array [
            "Banish all bast**ds",
          ],
        }
      }
      name="KP"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "sonys.jpg",
          "oath": Array [
            "Youcard is MINE!",
          ],
        }
      }
      name="Sonys"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "timo.jpg",
          "oath": Array [
            "Listen to TA",
            "Stop global warming",
          ],
        }
      }
      name="Timo"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "tim-armstrong.jpg",
          "oath": Array [
            "Give up!",
            "Take private jets",
          ],
        }
      }
      name="Tim Armstrong"
    />
    <PersonOathListItem
      info={
        Object {
          "image": "tsai-ing-wen.jpg",
          "oath": Array [
            "No more blackouts",
          ],
        }
      }
      name="Tsai Ing-wen"
    />
  </ul>
</div>
`;
